test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -909,6 +926,136 @@ TEST_P(ExtProcIntegrationTest, GetBufferedButNoBodies) {
   verifyDownstreamResponse(*response, 200);
 }
 
+TEST_P(ExtProcIntegrationTest, RemoveContentLengthInRequestHeader) {
```

Comment:
It feels a bunch of these cases should also be in unit tests.

Reply:
yea, many of unit tests are header only processing, so the content_length removal which is only enabled when body mutation happens is not applicable there. 

I added a few test cases to unit tests and I think the test coverage in integration test is quite comprehensive.